### ISSUE TITLE
DSC 6.5.3 cannot log to ClearingHouse

### ISSUE THREAD

**Issue by sebplorenz** [FraunhoferISST/DataspaceConnector#885]
When the DSC registers a new process id at the ClearingHouse, it sends an IDS message with payload (connector id of consumer and producer) to the clearing house. Like in this message:

```
--8b712f61-36b6-4cc2-ae5c-72c21c4fd933
Content-Disposition: form-data; name="header"
Content-Length: 2383

{
  "@context" : {
    "ids" : "https://w3id.org/idsa/core/",
    "idsc" : "https://w3id.org/idsa/code/"
  },
  "@type" : "ids:RequestMessage",
  "@id" : "https://w3id.org/idsa/autogen/requestMessage/cb79553f-fb53-40d1-9253-850a1f299740",
  "ids:issuerConnector" : {
    "@id" : "https://provider"
  },
  "ids:recipientConnector" : [ {
    "@id" : "http://tc-core:9999/process/b3f92843-7591-45dd-88be-73748148511f"
  } ],
  "ids:senderAgent" : {
    "@id" : "https://provider"
  },
  "ids:recipientAgent" : [ ],
  "ids:securityToken" : {
    "@type" : "ids:DynamicAttributeToken",
    "@id" : "https://w3id.org/idsa/autogen/dynamicAttributeToken/f1ee0ca3-d1c0-417b-af03-f41737b26965",
    "ids:tokenValue" : "...",
    "ids:tokenFormat" : {
      "@id" : "https://w3id.org/idsa/code/JWT"
    }
  },
  "ids:modelVersion" : "4.1.0",
  "ids:issued" : {
    "@value" : "2022-02-03T09:25:06.417Z",
    "@type" : "http://www.w3.org/2001/XMLSchema#dateTimeStamp"
  }
}
--8b712f61-36b6-4cc2-ae5c-72c21c4fd933
Content-Disposition: form-data; name="payload"
Content-Length: 50

{"owners":["https://provider","https://consumer"]}
--8b712f61-36b6-4cc2-ae5c-72c21c4fd933--
```
But instead of the connector URIs that are sent, the ClearingHouse expects the token client Ids and keys of the connectors.

FYI @kragall and @ronjaquensel 

Additionally, the payload part of the IDS message needs Content-Type: application/json to be set.

**Comment by HeinrichPet** [comment]
The old implementation of the Clearing House apparently did not check the payload, which is why it has worked for many who have tested it so far. The current implementation of the clearing house expects the [IDS-G-pre/Unique Identifier](https://github.com/International-Data-Spaces-Association/IDS-G-pre/blob/main/Components/IdentityProvider/DAPS/README.md#unique-identifier) as payload to identitfy the involved connectors.  However, this does not correspond to the newer definition on connector identification ([IDS Think Tank/Connector identity concept](https://github.com/International-Data-Spaces-Association/IDS-ThinkTank/blob/main/connector-identity/concept.md)), which is implemented by the DSC. We will clarify this further next week.

**Comment by ghost** [comment]
**Provider (Connector sending to the ClearingHouse):**
The `ski:keyid:aki` identifier combination, called by us `connector fingerprint` in the code, is also currently used internally by the DSC [for requesting the DAT from a DAPS](https://github.com/International-Data-Spaces-Association/IDS-Messaging-Services/blob/e038df1fafc5a47691b0b4e1ff58cc1b65b220ea/core/src/main/java/de/fraunhofer/ids/messaging/core/daps/aisec/AisecTokenManagerService.java#L392), as it is part of the request to the DAPS to identify the connector by its used certificate ([see: Request token that is handed in at DAPS side -> sub and iss](https://github.com/International-Data-Spaces-Association/IDS-G-pre/blob/main/Components/IdentityProvider/DAPS/README.md#request-token-that-is-handed-in-at-daps-side)). Due to the lack of clarity as to what the unique identifier of a connector actually is, currently there seems to be several at different levels, so we refer to this combination as the "fingerprint of the connector". [Code at which the fingerprint is determined](https://github.com/International-Data-Spaces-Association/IDS-Messaging-Services/blob/e038df1fafc5a47691b0b4e1ff58cc1b65b220ea/core/src/main/java/de/fraunhofer/ids/messaging/core/daps/aisec/AisecTokenManagerService.java#L411).

**Consumer**:
The `ski:keyid:aki` identifier of this connector is part of the DAT payload of the messages that this connector sends to us ([see: Dynamic Attribute Token Content -> payload fields -> sub field](https://github.com/International-Data-Spaces-Association/IDS-G-pre/blob/main/Components/IdentityProvider/DAPS/README.md#dynamic-attribute-token-content)): `...This is the connector requesting the DAT. The sub value must be the combined entry of the SKI and AKI of the IDS X509...` -> example: `"sub": "A5:0C:A5:F0:84:D9:90:BB:BC:D9:57:3A:04:C8:7F:93:ED:97:A2:52:keyid:CB:8C:C7:B6:85:79:A8:23:A6:CB:15:AB:17:50:2F:E6:65:43:5D:E8"`

**Other:**
Due to the new specification (draft) there could be braking changes when requesting a DAT from the DAPS (quote: "This should lead to a requirement that each connector MUST be identified by a unique hostname"), perhaps we should think about downward compatibility in this area, since [sub and iss in the request to the DAPS](https://github.com/International-Data-Spaces-Association/IDS-G-pre/blob/main/Components/IdentityProvider/DAPS/README.md#request-token-that-is-handed-in-at-daps-side) are both currently the `ski:keyid:aki` connector identifier and not any hostname. If this is not adopted for the DAT request, we will continue to have different identifiers of a connector and thus possible misunderstandings which identifier is actually needed, like in this Issue case.

**Current status:**
Currently it seems to be bilaterally agreed between the components what the identifier of a connector is.
- DSC <-> DSC: `URI`
- DSC <-> Broker: `URI`
- DSC <-> DAPS: `ski:keyid:aki`
- DSC <-> ClearingHouse: apparently also `ski:keyid:aki`

I could also have made a mistake in this short overview.

**For this Issue:**
If indeed `ski:keyid:aki` is needed for both connectors, all information is already available. Could probably be changed. The question is of course, how sustainable this solution then is, since the specification draft could change the identifier (globally).

**Comment by gbrost** [comment]
We discussed this a for some time. For the time being, we would suggest to stickt to aki/ski, since this is according to spec and it is available in all rolled out certificates, in contradiction to the Subject Alternative Name (SAN), which is not set everywhere etc. We could roll out certificates but this would be fragile and the implementation would be in limbo.
For a perspective: We also discussed going to the direction of URL as identifiers. But we get contradicting messages from different parties. We would have to define the deployment processes first, since:

- Now, you can request an id and edit als other attributes, including the self-description endpoint (referringConnector in DAT) later. 
- If we keep both an connector identity and separated TLS certificate, this would be a stable solution and we could change DNS entries later. 
- The identifier is not bound to a fixed protocol and not bound to DNS, which is not considered to be secure enough for some applications.

- If we go for a global identifier later, we should refrain from binding to a fixed protocol and use something more generic such as a URN scheme: urn:ids:someconnector.fraunhofer.de

TL;DR: We should stick to aki/ski. I hope the implementation/change burden is very low.

**Comment by ghost** [comment]
**Own connector:**
Adding a aki/ski fingerprint provider to the Messaging-Services to easily get the aki/ski of the own connector, see Changelog next version: https://github.com/International-Data-Spaces-Association/IDS-Messaging-Services/blob/development/CHANGELOG.md

**Comment by ghost** [comment]
**Other Connector:**
Getting the aki/ski of the Connector of the incoming-message-DAT is already supported by using the Messaging-Services `MessageAndClaimsHandler` instead of just the `MessageHandler`, which will also pass the Claims of the incoming DAT to the handler.

e.g. in my testing connector looks like this for incoming DescriptionRequestMessage as example:

![image](https://user-images.githubusercontent.com/75306992/154453697-926cb698-f7d3-4816-8b4c-2755798042b0.png)

printed log info (abbr. aki/ski):
`DAT SUB: 38:A1:F4:D1:37:16:...:keyid:CB:8C:C7:B6:85:79:...`

**Comment by ghost** [comment]
Will release a new Messaging-Services version today. 

Would mean for DSC:
- **own aki/ski (if available)**: ConnectorFingerprintProvider.fingerprint
- **other connector aki/ski (if available)**: MessageAndClaimsHandler -> Optional Claims -> body.subject

Would then have to be adopted where necessary.
